c/driver/postgresql/postgres_type.h

```diff
@@ -111,7 +111,11 @@ enum class PostgresTypeId {
   kXid8,
   kXid,
   kXml,
-  kUserDefined
+  kUserDefined,
+  // This is not an actual type, but there are cases where all we have is an Oid
+  // that was not inserted into the type resolver. We can't use "unknown" or "opaque"
+  // or "void" because those names show up in actual pg_type tables.
+  kUnnamedArrowOpaque
```

Comment:
Hmm, that seems like it should be an error? (or if we had warnings, a warning that you should recreate the connection to reload the OIDs)

Reply:
Eventually we'll probably have to tackle it as part of https://github.com/apache/arrow-adbc/issues/1755